runtime/common/src/main/kotlin/kotlinx/serialization/json/JsonConfiguration.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlinx.serialization.*
+import kotlin.jvm.*
+
+/**
+ * The class responsible for JSON-specific customizable behaviour in [Json] format.
+ *
+ * Options list:
+ * [encodeDefaults] specifies whether default values are encoded.
+ * [strictMode] enables strict mode, which prohibits unknown keys in the JSON and non-numerical values in floating point numbers.
+ * [unquoted] specifies whether keys and values should be quoted;
+ * [prettyPrint] specifies whether resulting JSON should be pretty-printed.
+ * [indent] specifies indent string to use with [prettyPrint] mode.
+ * [useArrayPolymorphism] switches polymorphic serialization to the default array format.
+ * [classDiscriminator] name of the class descriptor property in polymorphic serialization.
+ */
+public class JsonConfiguration(
```

Comment:
Btw, this constructor has also some kind of unstable behavior if we will add new flags which turn off some features by default (e.g. `allowNonTrivialKeysInMaps = false`). Is there any possible way to avoid it or this is inevitable evil?